diff --git a/bin/cli.py b/bin/cli.py
index b42839a..edb9025 100644
--- a/bin/cli.py
+++ b/bin/cli.py
@@ -1098,7 +1098,10 @@ def worker(args):
     }
 
     if conf.has_option("celery", "pool"):
-        options["pool"] = conf.get("celery", "pool")
+        user_pool = conf.get("celery", "pool")
+        options["pool"] = user_pool
+        from celery import maybe_patch_concurrency
+        maybe_patch_concurrency(['-P', user_pool])
 
     if args.daemon:
         pid, stdout, stderr, log_file = setup_locations("worker",
